### DISCUSSION TITLE
Support for executing multiple SQL queries in a single file

### DISCUSSION THREAD

**Discussion by dpnishant** [pawelsalawa/sqlitestudio#4522] [General]
### Details
Support for executing multiple SQL queries in a single `.sql` file separated by a semicolon. This shall be very useful for executing batch queries like database, tables, views, rows creation.

### Steps to reproduce
In version `3.3.3`, loading a `.sql` file with more than one queries (each separated by a semicolon `;`), only executes the first query.

**Comment by pawelsalawa** [comment]
You can do it by right-clicking on a database and picking "Execute SQL from file".

**Reply by dpnishant** [reply]
Thanks it worked. The `Load SQL from file` (toolbar button) in the SQL editor and `Execute SQL from file` context menu option seems to be confusing. It would be great if some obvious distinction could be added to these two separate options or better if both of them could be merged into one.

**Reply by dpnishant** [reply]
May be I misunderstood. `Load SQL from file` might be just meant for load the sql file in the editor and not execute it. In that case, the execute button in the editor needs to handle the execution of multiple queries versus executing only one at a time.

**Reply by pawelsalawa** [reply]
Sorry. My bad. Above I meant `Execute SQL from file`.

**Reply by pawelsalawa** [reply]
Regarding execution of single or multiple queries from the SQL Editor contents - it is done in a similar way as in some other, similar software. I mean - by default - only single query is executed (the one under the insertion cursor), but you can select any range of SQL contents and then hit `Execute SQL` to execute what you selected (including multiple queries at once).

Alternatively, you can change (in Configuration dialog) this default behavior, so that by all queries will be always executed by default.

**Reply by dpnishant** [reply]
Thanks for great explanation. :)

**Reply by DougBowen72** [reply]
@pawelsalawa Am I missing a setting or something? When I have multiple select statements in the editor and highlight them all and click execute it only returns results from the last query. I did uncheck "Execute only the query under the cursor" in the config dialog but that didn't make any difference. Thanks.

**Reply by pawelsalawa** [reply]
They are all executed, but you will see only results of the last statement.

**Reply by pawelsalawa** [reply]
If you have a series of INSERTs and then SELECT, you will see they all get executed. If you have series of SELECTs, they will also get all executed, but results from all of them will be disregarded, except for the last - which is displayed to user.

**Reply by DougBowen72** [reply]
Thanks for the clarification. So I suppose there's no way to view multiple result sets? Sometimes that's useful.

**Reply by pawelsalawa** [reply]
No. Still, you can just execute them in another SQL Editor subwindow.
